src/plugin-host/plugin-chat-services/chat-services.ts

```diff
@@ -0,0 +1,59 @@
+import { Chat } from "../../chat/chat";
+import { Leaderboard } from "../../chat/leaderboard/leaderboard";
+import { DankTime } from "../../dank-time/dank-time";
+import { User } from "../../chat/user/user";
+
+/**
+ * Plugin to Chat interface.
+ * Used by plugins to request certain data from their
+ * respective chats.
+ */
+export class ChatServices 
```

Comment:
Is this class needed? If we presume a server admin will only ever add plugins to his danktimesbot instance that are trusted, then why should we try and protect Chats from these plugins?

Reply:
Because abstracting away the internal state from any plugins also means that making changes to this internal state or the way that its internals work is easier to do without breaking the plugin API.